### PR TITLE
Test import layer and modularization

### PR DESCRIPTION
**Pull request by resetreboot**
`This is for review only, is still a WIP`

### CODE REVIEW

**Comment by resetreboot** [comment]
@carlosescri This is RTR.

File: packages/card-list/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export default {
+    input: './src/card-list.ts',
+    output: [
+      { file: pkg.module , format: 'esm' }
+    ],
+    external: [
+      '@dfwc/core',
+      '@dfwc/widget',
+      'lit-element',
+      'lit-html/directives/repeat',
```
Comment (carlosescri) [inline packages/card-list/rollup.config.js]: **Just to know:** what was the problem with this? Was not enough to declare `lit-html` as external? Was Rollup bundling the directive too? A comment may be necessary if the reason is not clear :wink:

Comment (resetreboot) [reply packages/card-list/rollup.config.js]: Sorry! From the rollupjs.org webpage, the `external ` option says:

```
Either a function that takes an id and returns true (external) or false (not external), or an Array of module IDs that should remain external to the bundle. The IDs should be either:

the name of an external dependency, exactly the way it is written in the import statement. I.e. to mark import "dependency.js" as external, use "dependency.js" while to mark import "dependency" as external, use "dependency".
a resolved ID (like an absolute path to a file).
```

So as the directive is imported as `lib-html/directives/repeat`, I just put there to make sure it remains as an external directive.

File: packages/card-list/src/card.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    if (+new Date() - this.mouseDownAt < (this.constructor as typeof Card).clickDuration) {
+      // first of all trigger our custom event so it can be intercepted ASAP
+      this.dispatchEvent(
+        replaceEvent(event, 'df-card-click', this, {
+          data: this.data.raw,
+        })
+      );
+
+      // dom access, can't cache it
+      const cardLink: HTMLElement = this.shadowRoot.getElementById('cardLink');
```
Comment (carlosescri) [inline packages/card-list/src/card.ts]: **Mental Note + @resetreboot:** `@query` and `@queryAll` to create props that perform `querySelector` and `querySelectorAll` on `this.renderRoot` (usually `this.shadowRoot`). I'm annotating this as an enhancement to be done when passing again through these files.

**Pull request merged**